arangod/RocksDBEngine/RocksDBEngine.cpp

```diff
@@ -795,8 +800,19 @@ void RocksDBEngine::start() {
   _options.compaction_readahead_size =
       static_cast<size_t>(opts._compactionReadaheadSize);
 
+  if (_createShaFiles) {
+    _checksumEnv = NewChecksumEnv(rocksdb::Env::Default(), _path);
+    _options.env = _checksumEnv.get();
+    static_cast<checksum::ChecksumEnv*>(_checksumEnv.get())
+        ->getHelper()
+        ->checkMissingShaFiles();
```

Comment:
Please leave a comment that this works despite not having set up the encryption yet (encryption setup follows in line 814 via the call to `configureEnterpriseRocksDBOptions()`).
This is because the `checkMissingShaFiles()` does not depend on the env or the encryption properly working, as the method will only look for existing .sst files and calculate file checksums from the file contents, unaware of any encryption etc.